packages/api/endpoints/granules.js

```diff
@@ -140,7 +141,7 @@ const create = async (req, res) => {
       granule.published = false;
```

Comment:
Is it necessary to set default value for published and createdAt?  The default values are also set at `packages/api/lib/writeRecords/write-granules.js:writeGranuleFromApi`

Reply:
also, _setNewGranuleDefaults can be used to set default values

Reply:
re: 

> Is it necessary to set default value for published and createdAt? The default values are also set at packages/api/lib/writeRecords/write-granules.js:writeGranuleFromApi

I think so, this is setting an expectation for API writes. 

If we're talking about the block that reads 

```
    const publishedValue = isNull(published) ? false : published;
    const defaultSetError = isNull(error) ? {} : error;
    const defaultSetFiles = isNull(files) ? [] : files;
    const defaultCreatedAt = isNull(createdAt) ? Date.now() : createdAt;
```

That's only for explicit nulls.      In this case, the logic is stating that in case of undefined *or* null we want to set a default.       We're making nulls a valid value for new granules to make the objects passed in more consistent.    Functionally nulls are the same as undefined in calls that create a new record. 

I went ahead and updated this code to use the newGranuleDefaults method and refactored that just a bit:  5ffd1a8ca3